### PR TITLE
Prerelease/qa one config

### PR DESCRIPTION
**Pull request by mshanemc**
> this is part of a set of PRs.  See https://salesforce.quip.com/kRtwAJN99VYE for explanation, sequence, and QA 

### What does this PR do?

- use the unified ConfigAggregator and Config
- that warns instead of throwing on deprecatedKeys, so much of the try/catch logic and NUTs had to change
- where Config.set didn't return useful information about how it translates old to new names (it return all the properties so you can't tell what you changed) it was previously returning the name passed in.  The plugin command now determines the "new" value if necessary before calling Config.set
- a new mixedNuts test that's mixing old and new keys to check for correct usage and prevent duplication between the two 
- adds plugin-deploy-retrieve as a dev-dep since some of the tests are using a config-meta that resides there and is not known to sfdx-core.  This required a change to the hooks UT
- set all the nuts to not do hubAuth

### What issues does this PR fix or reference?
@W-13048680@

### CODE REVIEW

File: src/commands/config/set.ts [hunk 1]
```diff
@@ -42,6 +42,8 @@ export class Set extends ConfigCommand<SetConfigCommandResult> {
 
     for (const name of Object.keys(parsed)) {
       const value = parsed[name] as string;
+      const resolvedName = this.configAggregator.getPropertyMeta(name)?.newKey ?? name;
+      // const resolvedName = Config.getAllowedProperties().find((p) => p.key === name)?.newKey ?? name;
```
Comment (WillieRuemmele) [inline src/commands/config/set.ts:46]: extra now

**Review by WillieRuemmele — APPROVED** [review]

**Comment by WillieRuemmele** [comment]
### QA Notes
---
setup
```
sfdx plugins install @salesforce/plugin-settings@qa

sfdx plugins --core
...
settings 1.5.0-qa.3 (qa)
```
---
❌ : thinks I have env vars set, suggests two similar vars... (doesn't happen with `sf`)
```
(node:64001) Warning: Deprecated environment variable: SFDX_AUTOUPDATE_DISABLE. Please use SF_AUTOUPDATE_DISABLE instead.
(Use `node --trace-warnings ...` to show where the warning was created)
(node:64001) Warning: Deprecated environment variable: SFDX_DISABLE_AUTOUPDATE. Please use SF_DISABLE_AUTOUPDATE instead.
```
I don't have either of those set
```
➜  dreamhouse-lwc git:(main) ✗  hub:(DevHub) scratch:(test-1a1u5sczeefj@example.com)
 ➜  env | grep SFDX              
➜  dreamhouse-lwc
```

✅ : `config set <deprecated>` suggests/displays new
✅ : `config get <deprecated>` shows correct, warns
✅ : `config get` a value that's not set
✅ : `config get` a deprecated value not set
✅ : `config get` an env var with deprecated config name
✅ : `config get` an env var with current name
❓ : should `config unset` print old name?
```
 ➜  sfdx config unset defaultdevhubusername   
Warning: Deprecated config name: defaultdevhubusername. Please use target-dev-hub instead.
Unset Config
===============================
| Name                  Success 
| ───────────────────── ─────── 
| defaultdevhubusername true    
```

❌ : does not respect the `SFDX_REST_DEPLOY` env var like others

- doesn't show up in `config list` as an environment

✅ : env var warnings in every command
```
 ➜  sfdx force data soql query -q "SELECT MemberType, MemberName, RevisionCounter FROM SourceMember WHERE MemberType='PermissionSet'" -t
(node:76699) Warning: Deprecated environment variable: SFDX_MAX_QUERY_LIMIT. Please use SF_ORG_MAX_QUERY_LIMIT instead.
```
💡 : what do you think about a new command, `migrate config` that moves config/envs to clear these warnings
❌ : `sf` ignoring `sfdx` env vars... seems to just be related to the "rest" config/env
```
 ➜  env | grep SF
SFDX_REST_DEPLOY=true
SFDX_MAX_QUERY_LIMIT=1
➜  dreamhouse-lwc git:(main) ✗  hub:(GLOBAL - DevHub) scratch:(test-1a1u5sczeefj@example.com)
 ➜  sf force source deploy -p force-app
Deploying v57.0 metadata to test-1a1u5sczeefj@example.com using the v57.0 SOAP API.
---
 ➜  env | grep SF
SFDX_MAX_QUERY_LIMIT=1
SF_ORG_METADATA_REST_DEPLOY=true
➜  dreamhouse-lwc git:(main) ✗  hub:(GLOBAL - DevHub) scratch:(test-1a1u5sczeefj@example.com)
 ➜  sfdx force source deploy -p force-app
Deploying v57.0 metadata to test-1a1u5sczeefj@example.com using the v57.0 SOAP API.
```
❌ : warnings only when using `sfdx`
```
 ➜  sf config list                       
List Config
=============================================
| Name                     Value  Location    
| ──────────────────────── ────── ─────────── 
| org-max-query-limit      1      Environment 
| org-metadata-rest-deploy true   Environment 
| target-dev-hub           DevHub Global      
➜  dreamhouse-lwc git:(main) ✗  hub:(GLOBAL - DevHub) scratch:(test-1a1u5sczeefj@example.com)
 ➜  sfdx config list                     
(node:80799) Warning: Deprecated environment variable: SFDX_MAX_QUERY_LIMIT. Please use SF_ORG_MAX_QUERY_LIMIT instead.
```

❌ : no more, "did you mean" interactivity?
```
 ➜  sfdx config set defaulttargetusername=test-1a1u5sczeefj@example.com
Error (1): Unknown config name: defaulttargetusername.
```

❌ : `sf` won't warn about deprecated env vars - maybe just related to the rest deploy - nope...
```
 ➜  export SFDX_REST_DEPLOY=true             
➜  dreamhouse-lwc git:(main) ✗  hub:(GLOBAL - DevHub) scratch:(test-1a1u5sczeefj@example.com)
 ➜  sf config list
List Config
============================================================

---
 ➜  export SFDX_MAX_QUERY_LIMIT=1 
➜  dreamhouse-lwc git:(main) ✗  hub:(GLOBAL - DevHub) scratch:(test-1a1u5sczeefj@example.com)
 ➜  sf config list                            
List Config
```

**Comment by WillieRuemmele** [comment]
✅ : unsets with old name and prints new
```
➜  sf config unset maxQueryLimit   
Warning: Deprecated config name: maxQueryLimit. Please use org-max-query-limit instead.
Unset Config
=============================
| Name                Success 
| ─────────────────── ─────── 
| org-max-query-limit true    
```
🏆 ✅ : interactivity is back
```
➜  dreamhouse-lwc git:(main) ✗  hub:(GLOBAL - DevHub) scratch:(test-1a1u5sczeefj@example.com)
 ➜  sf config set defaultdevusern=test
? Did you mean disable-telemetry? No
```

✅ : `sf` warns about deprecated env vars, I know you're looking at this though
```
 ➜  SFDX_MAX_QUERY_LIMIT=3 sf config list          
(node:89887) Warning: Deprecated environment variable: SFDX_MAX_QUERY_LIMIT. Please use SF_ORG_MAX_QUERY_LIMIT instead.
```

❌ : `plugin-source` might need to be updated
```
 ➜  sf config set restDeploy=true       
Warning: Deprecated config name: restDeploy. Please use org-metadata-rest-deploy instead.
Set Config
========================================
| Name                     Value Success 
| ──────────────────────── ───── ─────── 
| org-metadata-rest-deploy true  true    
➜  dreamhouse-lwc git:(main) ✗  hub:(GLOBAL - DevHub) scratch:(test-1a1u5sczeefj@example.com)
 ➜  sfdx force source deploy -p force-app                                                              
Warning: We plan to deprecate this command in the future. Try using the "project deploy start" command instead.
Error (1): Unknown config name: org-metadata-rest-deploy.
➜  dreamhouse-lwc git:(main) ✗  hub:(GLOBAL - DevHub) scratch:(test-1a1u5sczeefj@example.com)
 ➜  sf force source deploy -p force-app  
Warning: We plan to deprecate this command in the future. Try using the "project deploy start" command instead.
Error (1): Unknown config name: org-metadata-rest-deploy.
```

✅ : `org-max-query-limit` respected in every command, with appropriate warnings and deprecation messages
✅ : no warnings about auto-update env vars anymore when running from `sfdx-cli`
```
➜  sfdx-cli git:(sm/unified-config-aggregator) 
 ➜  ./bin/run config:list                 
List Config
================================
| Name           Value  Location 
| ────────────── ────── ──────── 
| target-dev-hub DevHub Global   
```

**Comment by mshanemc** [comment]
closed in favor of #248

**Pull request closed without merging**